### DISCUSSION TITLE
copy has some error

### DISCUSSION THREAD

**Discussion by opensofte** [umijs/umi#11251] [Q&A] [answered]
When I copied the files in node_modules, I was told they couldn't be located
  copy:[
    {
      from: 'D:\\project\\webstorm\\umi-react\\node_modules\\mediainfo.js\\dist\\MediaInfoModule.wasm\n',
      to: 'D:\\project\\webstorm\\umi-react\\'
    },
  ],
 I used the npm start command and was told something was wrong,which is :
 [
ERROR in unable to locate 'D:\project\webstorm\umi-react\D:\project\webstorm\umi-react\node_modules\mediainfo.js\dist\MediaInfoModule.wasm
' glob
]

**Comment by fz6m** [comment]
给一个最小复现看看。

**Reply by opensofte** [reply]
const wasmFile = resolve(
  __dirname,
  'node_modules',
  'mediainfo.js',
  'dist',
  'MediaInfoModule.wasm'
);

const devMode = process.env.NODE_ENV !== 'production'

const dist = devMode ? '.' : resolve(__dirname);

export default defineConfig({
  nodeModulesTransform: {
    type: 'none',
  },
  routes:routes,
  copy:[
    {
      from: '\\node_modules\\mediainfo.js\\dist\\MediaInfoModule.wasm',
      to: dist
    },
  ],
});

As mentioned above, when I describe this, I can locate the file I want
D:\project\webstorm\umi-react\node_modules\mediainfo.js\dist\MediaInfoModule.wasm

I wonder why this copy automatically carries dirname = D:\project\webstorm\, which is the parent folder of the project

**Reply by fz6m** [reply]
如有问题，需要提供一个可以复现的 repo 链接

**Comment by xiaohuoni** [comment, marked as answer]
https://umijs.org/docs/api/config#copy
> 你也可以通过对象配置具体的拷贝位置，其中相对路径的起点为项目根目录：

You can also configure specific copy locations through objects, where the starting point of the relative path is the project root directory:
